### ISSUE TITLE
RuntimeError: Exporting the operator col2im to ONNX opset version 12 is not supported

### ISSUE THREAD

**Issue by groovemaxRong** [KinWaiCheuk/nnAudio#102]
hi, i convert the stft model to onnx version , but a error is raised.
'''
RuntimeError: Exporting the operator col2im to ONNX opset version 12 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub
'''

**Comment by KinWaiCheuk** [comment]
At first glance, it seems this error is related to PyTorch version, rather than nnAudio. Which PyTorch version are you using?
Could you please share me the code for converting the stft model to onnx?
I will try to locate the exact problem, thanks!

**Comment by KinWaiCheuk** [comment]
I ran a quick test on my computer with the following code, and I can obtain the onnx file successfully.

```python
from nnAudio.Spectrogram import STFT
import torch

dummy_input = torch.randn(4,44100)
stft_layer = STFT()
torch.onnx.export(stft_layer,
                  dummy_input,
                  'stft_layer.onnx',
                  verbose=True,
                  input_names=['waveforms'],
                  output_names=['stft_output'])
```

I am using pytorch==1.9.0+cu111 and nnAudio==0.2.5. Could you double check if you are able to do ONNX using the same pytorch and nnAudio as me?

**Comment by groovemaxRong** [comment]
hi, i test it again and this problem might locate in the iSTFT module, here is the test code.
```
    from nnAudio.Spectrogram import STFT, iSTFT
    import torch
    
    class DummpyModel(nn.Module):
        def __init__(self):
            super(DummpyModel, self).__init__()
            self.stft_layer = STFT()
            self.istft_layer = iSTFT()
        
        def forward(self, wav):
            out = self.stft_layer(wav, output_format="Complex")
            out = self.istft_layer(out, length=wav.shape[-1], onesided=True)
            # out_wav = self.decoder(spec, wav.shape)
            return out

    dummy_input = torch.randn(4,44100)
    model = DummpyModel()
    torch.onnx.export(model,
                    dummy_input,
                    'stft_layer.onnx',
                    verbose=True,
                    opset_version=12,
                    input_names=['waveforms'],
                    output_names=['stft_output'])
```

**Comment by KinWaiCheuk** [comment]
You are right, when trying to export `iSTFT` to onnx, it will result in an error message:

> RuntimeError: Exporting the operator col2im to ONNX opset version 12 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.

This is because the `fold` function that iSTFT uses is not yet supported by onnx.

https://github.com/KinWaiCheuk/nnAudio/blob/ac9cd13315421dbc86582bed82359c2d4ba1ee6e/Installation/nnAudio/utils.py#L19-L29


There is a open issue under the PyTorch repo, I think people are working on it already.
https://github.com/pytorch/pytorch/issues/41423

Unfortunately, there is not much I can do at the moment. One workaround is to implement the two functions, `torch_window_sumsquare ` and `overlap_add ` without using `fold`. I do have an old version of nnAudio that did not use `fold` for inverse before commit [14db7a6e94e3f38ccaded7c779a85b99fdaefff1](https://github.com/KinWaiCheuk/nnAudio/commit/14db7a6e94e3f38ccaded7c779a85b99fdaefff1). But I remember the inverse was quite slow without using `fold`, and the result was slightly incorrect.

**Comment by veronicamorelli** [comment]
Operator col2im is supported for opset version 18 but not the previous ones, is they a way to overcome the error? 
"torch.onnx.errors.UnsupportedOperatorError: Exporting the operator 'aten::col2im' to ONNX opset version 17 is not supported. Support for this operator was added in version 18, try exporting with this version."
